### PR TITLE
Update browser compat docs.

### PR DESCRIPTION
**Pull request by arthurevans**
Take 1, updating all this stuff.

Staged here: https://3-0-polyfills-dot-polymer-project.appspot.com/3.0/docs/browsers

Most of the volume here is in my stab at adding info on modules.

Already really good stuff in Dan's README for the polyfills, unsure of whether to pull some of that in here, or to just point to it.

Also, it's still really hard to tell what browser needs what.
I wonder if it'd be better to just have a table like:

Browser & version range   | ES6/ES5 | ES Modules/AMD | Polyfills Req'd | Module name transform

Where we'd have multiple lines for a browser if the values were different.

(And yes, all browsers (currently) require the name=>path transform, but at least it makes it clear that some transform is always required.)

LMK what you think.

### CODE REVIEW

File: app/3.0/docs/es6.md [hunk 1]
```diff
@@ -1,46 +1,145 @@
...
+    [export](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export) pages on MDN 
+    provide a quick reference on module syntax.
+*   [Modules chapter](http://exploringjs.com/es6/ch_modules.html) from Axel Rauschmayer's Exploring ES6.
+
+As an addition to the standard ES6 module syntax, Polymer supports importing modules by name, as described
+in the next section.
+
+### Module specifiers
+
+The browser only accepts on kind of module specifier in an `import` statement: a URL, which must
```
Comment (katejeffreys) [inline app/3.0/docs/es6.md]: on -> one

File: app/3.0/docs/es6.md [hunk 2]
```diff
@@ -1,46 +1,145 @@
...
+Where `@polymer/polymer` is the _name_ of the npm package. (This style of specifier is sometimes called a 
+"bare module specifier".)
+
+These module specifiers need to be transformed to paths before they're served to the browser. The Polymer CLI can transform 
+them at build time, and the Polymer development server can transform them at runtime, so you can test code without a build step.
+Many third-party build tools, like WebPack and Rollup also support named modules.
+
+### Dynamic imports
+
+Dynamic imports is a language feature that enables lazy-loading of resources, replacing the functionality
```
Comment (katejeffreys) [inline app/3.0/docs/es6.md]: "Dynamic imports is a language feature"

is -> are? & if so, enables -> enable.

If not, maybe "Dynamic `import()` is a ..."

File: app/3.0/docs/es6.md [hunk 3]
```diff
@@ -1,46 +1,145 @@
...
+  console.log("MyView1 loaded");
+}).catch((reason) => {
+  console.log("MyView1 failed to load", reason);
+});
+```
+
+The latest versions of Chrome and Safari support dynamic imports. For other browsers, the Polymer build tools can transform
+dynamic imports into AMD modules. The Polymer development server can do this transformation on the fly.
+
+More on dyanmic import, see:
```
Comment (katejeffreys) [inline app/3.0/docs/es6.md]: More -> For more

"dyanmic" -> "dynamic"

"import" -> "imports" or "`import()`"

File: app/3.0/docs/es6.md [hunk 4]
```diff
@@ -1,46 +1,145 @@
...
-browser.
+browser. 
 
-**Device emulation may cause errors**. When using DevTools device emulation in Chrome,
-`polymer serve` will serve compiled code when emulating iOS devices, and uncompiled code when
-emulating Android devices. When switching between devices, the browser may end up with both
-compiled and uncompiled code in its cache, resulting in errors. To avoid this problem, run
-`polymer serve` with the `--compile never` option when testing with device emulation.
-{.alert .alert info}
+While the Polymer library doesn't depend on any features from newer JavaScript standards,
```
Comment (katejeffreys) [inline app/3.0/docs/es6.md]: This sentence sounds weird :)

File: app/3.0/docs/polyfills.md [hunk 5]
```diff
@@ -32,7 +28,7 @@ There are a couple of other related polyfill files that you may need:
     on browsers that support native custom elements. This is useful in static serving environments
     where you need to serve a single app version to all browsers. The adapter is discussed in more
     detail in [ES6](es6) and in [Build for production](/{{{polymer_version_dir}}}/toolbox/build-for-production).
-*   `apply-shim.html`. A polyfill for CSS mixins. Unlike the other polyfills, it should be included
+*   `apply-shim.j`. A polyfill for CSS mixins. Unlike the other polyfills, it should be imported
```
Comment (katejeffreys) [inline app/3.0/docs/polyfills.md]: .j -> .js

File: app/3.0/docs/polyfills.md [hunk 6]
```diff
@@ -130,10 +126,10 @@ Query parameter:
 
 ## Progress of native browser support
 
-As of April 2017, there has been broad cross-browser agreement around the v1 versions of the [Custom
+As of May 2018 the v1 versions of the [Custom
 Elements](https://w3c.github.io/webcomponents/spec/custom/) and [Shadow
-DOM](https://w3c.github.io/webcomponents/spec/shadow/) APIs, with support in Chrome, Opera, and
-Safari Tech Preview, and implementations underway in other browsers.
+DOM](https://w3c.github.io/webcomponents/spec/shadow/) APIs have been shipped in Chrome, Opera, and
+Safari, and implementations underway in other browsers.
```
Comment (katejeffreys) [inline app/3.0/docs/polyfills.md]: are underway

File: app/3.0/docs/polyfills.md [hunk 7]
```diff
@@ -4,18 +4,14 @@ title: Polyfills
...
 
-<div>
-{% include 'outdated.html' %}
-</div>
-
-Polymer 2.x has been developed alongside and tested with a new suite of v1-spec compatible polyfills
-for custom elements and shadow DOM. You can test Polymer 2.x by using the latest 1.0 version of
-`webcomponentsjs`, which is included as a bower dependency to Polymer 2.x. (`webcomponentsjs`
+Polymer 3.x has been developed alongside and tested with a new suite of v1-spec compatible polyfills
+for custom elements and shadow DOM. You can test Polymer 3.x using the latest 2.0 version of
```
Comment (katejeffreys) [inline app/3.0/docs/polyfills.md]: should we mention explicitly that the HTML Imports polyfill has been removed from v2.0 & there is no webcomponents-lite.js?

**Comment by katejeffreys** [comment]
i like the idea of a table showing exactly what polyfills & build transforms each browser requires, provided it wouldn't require too much maintenance for it to stay accurate.

File: app/3.0/docs/polyfills.md (hunk 7 above)
Comment (katejeffreys) [reply app/3.0/docs/polyfills.md]: & maybe even that you need to do this in polymer.json if using loader?

```
"extraDependencies": [
    "node_modules/@webcomponents/webcomponentsjs/bundles/**"
  ]
```

**Comment by web-padawan** [comment]
BTW the relevant quote from Polymer/tools#191

>
Modules support now requires dynamic import and import.meta.

Specifically, this means:

Safari modules support is now from 11.1 instead of 10.1/10.3
Chrome modules support is now from 64 instead of 61.
Vivaldi supports modules (tested).

**Pull request merged**